docs/guide/custom-functions.md

```diff
@@ -1,66 +1,59 @@
 
-As an example, let's create a function that returns the number of letters in the word "HyperFormula".
-
-<iframe
-  src="https://codesandbox.io/embed/github/handsontable/hyperformula-demos/tree/2.2.x/custom-functions?autoresize=1&fontsize=11&hidenavigation=1&theme=light&view=preview"
-  style="width:100%; height:300px; border:0; border-radius: 4px; overflow:hidden;"
-  title="handsontable/hyperformula-demos: custom-functions"
-  allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
-  sandbox="allow-autoplay allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts">
-</iframe>
+As an example, let's create a custom function `GREET` that accepts a person's first name as a string argument and returns a personalized greeting.
 
 ### 1. Create a function plugin
 
 Import `FunctionPlugin`, and extend it with a new class. For example:
 
-```javascript
+```js
 import { FunctionPlugin } from 'hyperformula';
 
-// let's call the function plugin `CountHF`
-export class CountHF extends FunctionPlugin {
+// let's call the function plugin `MyCustomPlugin`
+export class MyCustomPlugin extends FunctionPlugin {
   
 }
 ```
 
-### 2. Define your function's ID and method
+### 2. Define your function's ID, method, and metadata
 
-In your function plugin, in the static `implementedFunctions` property, define an object that contains your custom function.
+In your function plugin, in the static `implementedFunctions` property, define an object that declares the functions provided by this plugin.
 
-The name of that object becomes the ID by which [translations](#function-translations), [aliases](#function-aliases), and other elements refer to your function.
+The name of that object becomes the ID by which [translations](#function-name-translations), [aliases](#function-aliases), and other elements refer to your function.
 Make the ID unique among all HyperFormula functions ([built-in](built-in-functions.md#list-of-available-functions) and custom).
 
-Then, in your function's object, define a `method` property, which maps your function to an implementation method (we'll define it later on).
+In your function's object, you can specify:
+- a `method` property (required), which maps your function to the implementation method (we'll define it later on),
+- a `parameters` array that describes the arguments accepted by your function and [validation options](#argument-validation-options) for each argument,
+- other [custom function options](#custom-function-options).
 
-```javascript
-CountHF.implementedFunctions = {
-  // let's define the function's ID as `HYPER`
-  HYPER: {
-    // we'll define the `hyper` method later on
-    method: 'hyper',
-  },
+```js
+MyCustomPlugin.implementedFunctions = {
+  // let's define the function's ID as `GREET`
+  GREET: {
+    method: "greet",
+    parameters: [
+      { argumentType: "STRING" }
```

Comment:
In TS, the string literal does not work. It must be `FunctionArgumentType.STRING`. 

I think it should either be explained here or fixed in the types.

Reply:
Generally, for TS devs we should create separate example/demo/code snippet and describe the exported enums & interfaces, but I think it's out of the scope of this task so I wrote everything with plan JS in mind.

Reply:
You're right, once `FunctionArgumentType` is exported it can be used here. I'll change it.

Reply:
done